src/internal/types/TupleParts.ts

```diff
@@ -2,37 +2,53 @@ import type { IsEqual } from "type-fest";
 
 /**
  * Takes an array and returns the types that make up its parts. The prefix is
- * anything before the rest parameter (if any), the suffix is anything after
- * the rest parameter (if any), and the item is the type of the rest parameter.
+ * anything before the rest parameter (if any), split between it's required
+ * part, and it's optional part. The suffix is anything after the rest
+ * parameter (if any), and the item is the type of the rest parameter.
+ *
+ * NOTE: The prefix is split into 2 tuples where all items are non-optional so
+ * that types that rely on the presence of a specific element can be built more
+ * accurately. For backwards compatibility, the `prefix` accessor which hides
+ * some of this complexity is also provided (but we might deprecate it in the
+ * future).
```

Comment:
**thought:** i think we should just deprecate it, if it's not hard to

Reply:
It's more work than made sense for this PR; it requires looking into each case and seeing how the optionality changes it. When I built the type for chunk I made an assumption that the tuples don't have optional members.

I'll replace this with a TODO so we know this needs to happen.